letsencrypt-nginx/letsencrypt_nginx/nginxparser.py

```diff
@@ -1,56 +1,241 @@
-"""Very low-level nginx config parser based on pyparsing."""
-import string
+"""Very low-level hand rolled nginx config parser, using recursive descent to
+retain all parsed information, so that the file can be reconstituted."""
 
-from pyparsing import (
-    Literal, White, Word, alphanums, CharsNotIn, Forward, Group,
-    Optional, OneOrMore, Regex, ZeroOrMore)
-from pyparsing import stringEnd
-from pyparsing import restOfLine
+import StringIO
 
+class ParseException(Exception):
+    """A parsing exception happened."""
+    def __init__(self, value):
+        self.value = value
+        super(ParseException, self).__init__()
```

Comment:
There's no sense in calling the superclass's constructor if you're not passing anything to it. All it does is set `self.args`. I wouldn't bother making this call at all, since you clearly mean to use the `value` attr instead.